Make it possible for users to view their data on Panorama Public

vagisha opened the pull request:
#### Changes
- Added a "My Data" button in the "Panorama Public Experiments" button bar. The button is displayed if a user is logged in. It links to MyDataViewAction. 
![image](https://github.com/LabKey/MacCossLabModules/assets/76364/6e7f8a5a-1f9f-4e0d-938c-c907723d6735)

- My Data view displays a grid filtered to the user's public and private datasets on Panorama Public. These are experiments where the user is either a "submitter" or the "lab head". 
- The "My Data" grid displays a "Catalog Entry" column that can be used to add or view catalog entries for public datasets. Catalog entries cannot be added for private datasets.

![image](https://github.com/LabKey/MacCossLabModules/assets/76364/cc4c2311-42e7-47d6-a766-a49d708257f2)

vagisha committed:
Add a "My Data" view for datasets submitted to Panorama Public.

vagisha committed:
Added test.

vagisha committed:
Moved "CatalogEntry" column to ExperimentAnnotationsTableInfo.

vagisha committed:
Removed MyDataView

vagisha committed:
Fix query for catalog entry column

vagisha committed:
Use string constant.

labkey-jeckels commented on panoramapublic/src/org/labkey/panoramapublic/PanoramaPublicController.java:
```diff
@@ -9183,6 +9184,68 @@ public ModelAndView getView(Object o, BindException errors)
+        }
+    }
+
+    public static class MyDataForm
```
Do we need this form at all? It seems like we only support viewing data for the current user.

vagisha replied:
No, we don't need a userId since we are only displaying the data for the currently logged in user!  I removed the form.

labkey-jeckels commented on panoramapublic/src/org/labkey/panoramapublic/query/ExperimentAnnotationsTableInfo.java:
```diff
@@ -724,4 +747,47 @@ String getRenderId()
+                    return;
+                }
+            }
+            HtmlString.EMPTY_STRING.appendTo(out);
```
This codepath uses `EMPTY_STRING` and the one above uses `NBSP`. I imagine they should probably use the same placeholder, though maybe it doesn't make a difference.

vagisha replied:
Done.

vagisha committed:
- Code review changes (remove form in MyDataViewAction, use HtmlString.NBSP consistently)
- Use return URL, when available, for rendering the back button in the CatalogEntryWebPart.
